docs/user-guide/at-tls-configuration.md

```diff
@@ -1,23 +1,642 @@
+<!-- omit in toc -->
+# Configuring AT-TLS for Zowe Server
 
-The communication server on z/OS provides functionality to encrypt HTTP communication for on-platform jobs. This functionality is referred to as Application Transparent Transport Layer Security (AT-TLS).
+Review this article for descriptions of the configuration parameters required to make Zowe work with AT-TLS, including AT-TLS inbound and outbound rules, using AT-TLS in high availability, and troubleshooting. Security recommendations are also provided.
 
-:::info Required roles: security administrator
+:::info Role: security administrator
 :::
 
-## Configuration Parameters
+- [AT-TLS configuration for Zowe](#at-tls-configuration-for-zowe)
+  - [Limitations](#limitations)
+- [AT-TLS rules](#at-tls-rules)
+  - [Inbound rules](#inbound-rules)
+  - [Outbound rules](#outbound-rules)
+    - [For z/OSMF](#for-zosmf)
+    - [For communication between API Gateway and other core services](#for-communication-between-api-gateway-and-other-core-services)
+    - [For communication between API Gateway and southbound services](#for-communication-between-api-gateway-and-southbound-services)
+  - [Ciphers](#ciphers)
+- [Using AT-TLS for API ML in High Availability](#using-at-tls-for-api-ml-in-high-availability)
+- [Multi-tenancy deployment](#multi-tenancy-deployment)
+- [AT-TLS Troubleshooting](#at-tls-troubleshooting)
+  - [The message `This combination of port requires SSL` is thrown](#the-message-this-combination-of-port-requires-ssl-is-thrown)
+  - [AT-TLS rules are not applied](#at-tls-rules-are-not-applied)
+  - [Non matching ciphers](#non-matching-ciphers)
+- [Full example of AT-TLS configuration](#full-example-of-at-tls-configuration)
+- [See also](#see-also)
+
+## AT-TLS configuration for Zowe
+
+:::tip
+For full support, we recommend that you upgrade to v2.13 or a later version of Zowe.
+:::
+
+Follow these steps to configure Zowe to support AT-TLS:
+
+These instructions are valid for Zowe 2.18 onwards:
 
-To enable AT-TLS for Zowe components, configure the following parameters:
 ```yaml
 zowe:
     network:
+        # For inbound traffic rules:
         server:
             tls:
                 attls: true
+        # If outbound traffic rules will be configured:
+        client:
+          tls:
+            attls: true
+```
+
+While the Zowe Server components do not handle TLS on its own with AT-TLS enabled, the API Mediation Layer (API ML) requires information about the server certificate that is defined in the AT-TLS rule. Ensure that the server certificates provided by the AT-TLS layer are trusted in the configured Zowe keyring. Ideally, AT-TLS should be configured with the same Zowe keyring.
+
+If there is an outbound AT-TLS rule configured for the link between the API Gateway and z/OSMF, set the `zowe.zOSMF.scheme` property to `http`.
+
+:::note Notes
+
+- AT-TLS is supported in the API Cloud Gateway Mediation Layer component beginning with version 2.17.
```

Comment:
I guess Cloud Gateway is based on Netty, so aware mode is not supported.

Reply:
This is an old phrase in the doc, you are saying we should remove it? Or clarify the aware mode is not supported?

Reply:
I am not sure, probably, something like:

AT-TLS is supported in the API Cloud Gateway Mediation Layer component in Basic mode. It doesn't support client certificates and in the case of missing AT-TLS rule allows to communicate with HTTP protocol.